### PR TITLE
feat: strictly validate the input of core resources

### LINKED ISSUE #10229: bug: APISIX core resources require strict verification and do not need to pass in additional fields.
the original PR https://github.com/apache/apisix/pull/4797

### Current Behavior

```
# curl http://127.0.0.1:9180/apisix/admin/routes -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X POST -d '{
                       "upstream": {
                           "type": "roundrobin",
                           "nodes": {
                               "httpbin.org": 1
                           },
                           "vvv": "ccc"
                       },
                       "uri": "/get"
                   }'
                   
{"key":"/apisix/routes/00000000000000013083","value":{"upstream":{"type":"roundrobin","scheme":"http","pass_host":"pass","nodes":{"httpbin.org":1},"hash_on":"vars","vvv":"ccc"},"id":"00000000000000013083","status":1,"create_time":1694071341,"uri":"/get","update_time":1694071341,"priority":0}}
```

```
# etcdctl get /apisix/routes/00000000000000013083

/apisix/routes/00000000000000013083
{"id":"00000000000000013083","priority":0,"update_time":1694071341,"create_time":1694071341,"uri":"/get","status":1,"upstream":{"type":"roundrobin","scheme":"http","pass_host":"pass","nodes":{"httpbin.org":1},"hash_on":"vars","vvv":"ccc"}}
```

### Expected Behavior

No additional fields can be passed in core resources.

Core resources include

- [x] Consumer
- [x] Consumer Group
- [x] Global Rules
- [x] Plugin
- [x] Plugin Config
- [x] Route
- [x] stream route
- [x] Service
- [x] Upstream

### Error Logs

N/A

### Steps to Reproduce

N/A

### Environment

- APISIX version (run `apisix version`): master
- Operating system (run `uname -a`): ubuntu
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by Sn0rt**
### Description

Fixes #10229

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/wasm/global-rule.t [hunk 1]
```diff
@@ -57,13 +57,6 @@ __DATA__
             local code, body = t('/apisix/admin/global_rules/1',
                 ngx.HTTP_PUT,
                 [[{
-                    "uri": "/hello",
-                    "upstream": {
-                        "type": "roundrobin",
-                        "nodes": {
-                            "127.0.0.1:1980": 1
-                        }
-                    },
```
Comment (monkeyDluffy6017) [inline t/wasm/global-rule.t:66]: Why do you remove this?

Comment (Sn0rt) [reply t/wasm/global-rule.t:66]: This field is not as expected and is not allowed to be passed into upstream in the global-rule.

Comment (monkeyDluffy6017) [reply t/wasm/global-rule.t:66]: If you delete this block, the test case will fail

Comment (Sn0rt) [reply t/wasm/global-rule.t:66]: I have test this PR on myself repo https://github.com/Sn0rt/apisix/pull/9

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by AlinsRan — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**